### PR TITLE
fix directive sort compare

### LINKED ISSUE #3768: Directive params casted into strings
### Vue.js version
1.0.27

### Description
Parameter data type is illegal to chang

### Reproduction Link

[jsfiddle](https://jsfiddle.net/jawc4bpo/)

### What is Expected?

Here is same [example](https://jsfiddle.net/gn4aezn2/)  with version 1.0.26 where it works

### PR DESCRIPTION
**Pull request by defcc**
fix #3768 

Object.keys(groupedMap) returns an array whose elements are strings, so it breaks the sort function.

### CODE REVIEW

**Comment by posva** [comment]
Thanks! LGTM
I'm not sure about the test though

**Comment by defcc** [comment]
Ok, current test case seems not to fully cover the code. I will try to update the test case soon

File: src/compiler/compile.js [hunk 1]
```diff
@@ -122,18 +122,20 @@ function sortDirectives (dirs) {
...
     if (!array) {
       array = groupedMap[priority] = []
+      keys.push(priority)
     }
     array.push(dir)
   }
 
   var index = 0
-  var priorities = Object.keys(groupedMap).sort(function (a, b) {
+  var priorities = keys.sort(function (a, b) {
```
Comment (yyx990803) [inline src/compiler/compile.js]: sort is in-place, so we can just name this array `priorities` from the start, no need to assign a separate variable.

**Comment by defcc** [comment]
forgot it. I will fix it later today

File: src/compiler/compile.js (hunk 1 above)
Comment (defcc) [inline src/compiler/compile.js]: thanks, I have updated the code :)

**Comment by yyx990803** [comment]
Thanks! 👍

**Comment by juukie** [comment]
Thanks, fixed the bug 👍

**Pull request merged**